### ISSUE TITLE
[Bug]: ANR because of initializeFromJson

### ISSUE THREAD

**Issue by ayushi-sk** [OneSignal/OneSignal-Android-SDK#1996]
### What happened?

com.onesignal.common.modeling.Model.initializeFromJson
ANR triggered by slow operations in main thread

### Steps to reproduce?

```Markdown
Initialise OneSignal on android 11, 12 and 13
```

### What did you expect to happen?

No ANR please

### OneSignal Android SDK version

5.1.2 and all others

### Android version

13, 12, 11

### Relevant log output

```Shell
at kotlin.collections.ArraysKt___ArraysKt.firstOrNull(ArraysKt___Arrays.kt:1282)
       at com.onesignal.common.modeling.Model.initializeFromJson(Model.kt:98)
       at com.onesignal.common.modeling.SimpleModelStore.create(SimpleModelStore.kt:30)
       at com.onesignal.common.modeling.ModelStore.load(ModelStore.kt:162)
       at com.onesignal.common.modeling.SimpleModelStore.<init>(SimpleModelStore.kt:23)
       at com.onesignal.session.internal.session.SessionModelStore.<init>(SessionModelStore.kt:8)
       at java.lang.reflect.Constructor.newInstance0(Native method)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:89)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceProvider.getService$com_onesignal_core(ServiceProvider.kt:39)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:200)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:208)
       at com.onesignal.notifications.receivers.FCMBroadcastReceiver.onReceive(FCMBroadcastReceiver.kt:26)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by emawby** [comment]
@ayushi-sk Hello thank you for reaching out. Is this happening on versions newer than 5.1.2? Are you able to reproduce this issue?

**Comment by ayushi-sk** [comment]
Couldn't reproduce it -> happening on all device 
Yes, it is in 5.1.6

**Comment by gitByEOS** [comment]
updated to com.onesignal:OneSignal:5.1.9 also Appeared

```
#00 pc 0x4bca80 libart.so (art::DumpNativeStack + 108) (BuildId: 735f12f804f88d62a2cb437261076ff7)
#01 pc 0x4b0dd4 libart.so (art::Thread::DumpStack const + 388) (BuildId: 735f12f804f88d62a2cb437261076ff7)
#02 pc 0x4b0714 libart.so (art::DumpCheckpoint::Run + 164) (BuildId: 735f12f804f88d62a2cb437261076ff7)
#03 pc 0x3c177c libart.so (art::Thread::RunCheckpointFunction + 148) (BuildId: 735f12f804f88d62a2cb437261076ff7)
#04 pc 0x5aea28 libart.so (artTestSuspendFromCode + 232) (BuildId: 735f12f804f88d62a2cb437261076ff7)
#05 pc 0x35521c libart.so (art_quick_test_suspend + 156) (BuildId: 735f12f804f88d62a2cb437261076ff7)
       at java.lang.reflect.Executable.getMethodNameInternal(Native method)
       at java.lang.reflect.Method.getName(Method.java:115)
       at java.lang.reflect.Method$1.compare(Method.java:72)
       at java.lang.reflect.Method$1.compare(Method.java:67)
       at java.util.TimSort.mergeHi(TimSort.java:841)
       at java.util.TimSort.mergeAt(TimSort.java:520)
       at java.util.TimSort.mergeForceCollapse(TimSort.java:461)
       at java.util.TimSort.sort(TimSort.java:254)
       at java.util.Arrays.sort(Arrays.java:1344)
       at java.util.ArrayList.sort(ArrayList.java:1738)
       at java.util.Collections.sort(Collections.java:210)
       at libcore.util.CollectionUtils.removeDuplicates(CollectionUtils.java:86)
       at java.lang.Class.getMethods(Class.java:2226)
       at com.onesignal.common.modeling.Model.initializeFromJson(Model.kt:98)
       at com.onesignal.core.internal.operations.impl.OperationModelStore.create(OperationModelStore.kt:68)
       at com.onesignal.core.internal.operations.impl.OperationModelStore.create(OperationModelStore.kt:30)
       at com.onesignal.common.modeling.ModelStore.load(ModelStore.kt:162)
       at com.onesignal.core.internal.operations.impl.OperationModelStore.<init>(OperationModelStore.kt:32)
       at java.lang.reflect.Constructor.newInstance0(Native method)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:89)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:82)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:527)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:135)
       at org.xxxxxxxxxxxxxxxxxxx.initOneSignal(XXXXXXApplication.java:121)
       at org.xxxxxxxxxxxxxxxxxxx.onCreate(XXXXXXApplication.java:53)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1266)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7411)
       at android.app.ActivityThread.-$$Nest$mhandleBindApplication(unavailable)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2410)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:223)
       at android.os.Looper.loop(Looper.java:324)
       at android.app.ActivityThread.main(ActivityThread.java:8630)
       at java.lang.reflect.Method.invoke(Native method)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:582)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1061)
```

Android version
14 ~1%，13 ~31% , 12 ~31%, 11 ~37% 

Start from background 97% 
(data from firebase Crashlytics dashboard)

**Comment by jinliu9508** [comment]
@ayushi-sk @gitByEOS 
Thank you for sharing the valuable info with us! We have released Android SDK 5.1.12 in an attempt to resolve the specific ANR issues. Please give it a try and let us know if the ANR still persists. https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.12

**Comment by StrozhDima** [comment]
@jinliu9508 It's still happening with Android SDK version 5.1.12

`main (runnable):tid=1 systid=27054 
#00 pc 0x4fba2c libart.so (art::DumpNativeStack + 108) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#01 pc 0x4fe0e4 libart.so (art::Thread::DumpStack const + 376) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#02 pc 0x4ffed8 libart.so (art::DumpCheckpoint::Run + 216) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#03 pc 0x22c2b0 libart.so (art::Thread::RunCheckpointFunction + 144) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#04 pc 0x493c28 libart.so (art::ClassLinker::LoadClass + 10484) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#05 pc 0x2dd4e8 libart.so (art::ClassLinker::DefineClass + 2508) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#06 pc 0x686d4c libart.so (art::ClassLinker::FindClassInBaseDexClassLoader + 868) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#07 pc 0x24cd28 libart.so (art::ClassLinker::FindClass + 736) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#08 pc 0x22bcac libart.so (art::ObjPtr<art::mirror::Class> art::ClassLinker::DoResolveType<art::ArtMethod*> + 292) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
#09 pc 0x75c428 libart.so (art::Executable_getMethodReturnTypeInternal  + 76) (BuildId: a0856b3dbc826e8cee9d66738ac739d8)
       at java.lang.reflect.Executable.getMethodReturnTypeInternal(Native method)
       at java.lang.reflect.Method.getReturnType(Method.java:148)
       at com.onesignal.common.modeling.Model.initializeFromJson(Model.kt:99)
       at com.onesignal.common.modeling.SimpleModelStore.create(SimpleModelStore.kt:30)
       at com.onesignal.common.modeling.ModelStore.load(ModelStore.kt:175)
       at com.onesignal.common.modeling.SimpleModelStore.<init>(SimpleModelStore.kt:23)
       at com.onesignal.user.internal.subscriptions.SubscriptionModelStore.<init>(SubscriptionModelStore.kt:7)
       at java.lang.reflect.Constructor.newInstance0(Native method)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:89)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:82)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:82)
       at com.onesignal.common.services.ServiceProvider.getAllServices(ServiceProvider.kt:55)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:72)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:35)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:208)
       at com.onesignal.notifications.receivers.FCMBroadcastReceiver.onReceive(FCMBroadcastReceiver.kt:26)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:4497)
       at android.app.ActivityThread.-$$Nest$mhandleReceiver(unavailable)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2239)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:201)
       at android.os.Looper.loop(Looper.java:288)
       at android.app.ActivityThread.main(ActivityThread.java:8126)
       at java.lang.reflect.Method.invoke(Native method)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:588)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1019)
`

48% - Android 12
37% - Android 13
15% - Android 11

**Comment by dupontAlen** [comment]
Any update for this bug. It still happen even for the last build "5.1.31"

**Comment by ayushi-sk** [comment]
Yes, it is still happening!

No crashes has been resolved in the latest 5.1.31 version

**Comment by mariamikv** [comment]
Any update for this bug. Still reproducible in v5.7.6
Confirming this is not fixed. We're seeing the same ANR in production (via Sentry) on the latest release.
SDK version: 5.7.6
`Instrumentation.callApplicationOnCreate`
